grails-app/init/org/pih/warehouse/BootStrap.groovy

```diff
@@ -212,7 +212,7 @@ class BootStrap {
                     name       : inventoryItem?.product?.name,
                     productCode: inventoryItem?.product?.productCode
                 ],
-                lotNumber     : inventoryItem.lotNumber,
+                lotNumber     : inventoryItem.lotNumber ?: null,
```

Comment:
I know this isn't new from you, but I wonder why there is both InventoryItem.toJson() and this? Surely this should be:

```
JSON.registerObjectMarshaller(InventoryItem) { InventoryItem inventoryItem ->
    return inventoryItem.toJson()
}
```

Reply:
but more importantly, this is going to have implications anywhere inventory items are used. I wouldn't be surprised if this breaks the frontend elsewhere because it was expecting '' but is now getting null. This will need testing to make sure old flows still work.

Reply:
Interesting point. This might actually have an implicit coersion from "" to null if the "" evaluates to false within the elvis operator. 
```
inventoryItem.lotNumber ?: null
```

> I know this isn't new from you, but I wonder why there is both InventoryItem.toJson() and this? Surely this should be:

We probably missed registering some object marshallers. I would be happy to see this in 0.9.5 but adding it to the hotfix would be dangerous since it would change the JSON response anywhere we're returning an inventory item object.
```
JSON.registerObjectMarshaller(InventoryItem) { InventoryItem inventoryItem ->
    return inventoryItem.toJson()
}
```